src/librustc/diagnostics.rs

```diff
@@ -196,9 +196,13 @@ const Y: i32 = A;
 "##,
 
 E0015: r##"
-The only function calls allowed in static or constant expressions are enum
-variant constructors or struct constructors (for unit or tuple structs). This
-is because Rust currently does not support compile-time function execution.
+The only function calls allowed in static or constant expressions are
+constructors for enum variants and structs, or `const fn`s. Rust currently does
```

Comment:
The fact that enum variants and tuple structs are function calls is just confusing here. Now `const fn` is a thing, it's probably better to just say "only `const fn`s can be called in static or constant expressions, since they obviously would have seen this error by using a non-`const fn` call.